akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -414,7 +415,11 @@ import akka.util.ByteString
             handshakeFinished()
             transportInChoppingBlock.putBack(transportInBuffer)
           case _ =>
-            if (transportInBuffer.hasRemaining) doUnwrap(ignoreOutput = false)
+            if (transportInBuffer.hasRemaining)
```

Comment:
so this could be in state `NOT_HANDSHAKING`, `NEED_TASK`, `NEED_UNWRAP` or `NEED_UNWRAP_AGAIN`. It's not super obvious that it's an error when you end up in those states without advancing the transport-in-buffer...

Reply:
Good point, let's restrict this condition to `NEED_UNWRAP`.